### ISSUE TITLE
Empty string seems to remove previous text

### ISSUE THREAD

**Issue by rverberne** [QuestPDF/QuestPDF#906]
**Describe the bug**
The code produces a document containing:
```
AAA
```

**To Reproduce**
See full reproduction code below.

**Expected behavior**
A document containing:
```
AAA
BBB
```

**Screenshots**
<None>

**Environment**
2024.3.10
Windows

**Additional context**
A empty string or null will seemingly remove the previous added element. In this case the text `BBB` gets removed when the empty string in `GroupTwo` is present. Changing that string to `CCC` or something, then the document will contain `AAA`, `BBB` and `CCC`.

Also, when removing `column.Spacing(4, Unit.Point);` in `GroupTwo` then `BBB` comes back.

```csharp
using QuestPDF.Fluent;
using QuestPDF.Helpers;
using QuestPDF.Infrastructure;

internal class Program
{
    private static void Main(string[] args)
    {
        QuestPDF.Settings.License = LicenseType.Community;
        var doc = Document.Create(container =>
        {
            container.Page(page =>
            {
                page.Size(PageSizes.A4);
                page.Content()
                    .Column(column =>
                    {
                        column
                            .Item()
                            .Component(new GroupOne());
                    });
            });
        });

        doc.GeneratePdf("");
    }
}

public class GroupOne : IComponent
{
    public void Compose(IContainer container)
    {
        container
            .Column(column =>
            {
                column.Spacing(4, Unit.Point);
                column
                    .Item()
                    .Component(new PlainText("AAA"));
                column
                    .Item()
                    .Component(new PlainText("BBB"));
                column
                    .Item()
                    .Component(new GroupTwo());
            });
    }
}

public class GroupTwo : IComponent
{
    public void Compose(IContainer container)
    {
        container
            .Column(column =>
            {
                column.Spacing(4, Unit.Point);

                column
                    .Item()
                    .Component(new PlainText(""));
            });
    }
}

public class PlainText : IComponent
{
    private readonly string? s;

    public PlainText(string? s)
    {
        this.s = s;
    }

    public void Compose(IContainer container)
    {
        var text = container.Text(s as string);
    }
}
```

**Comment by MarcinZiabek** [comment]
Thank you for finding this issue and sharing the reproducible code 😄 

I have investigated this bug and found a root cause. The fix will be released next week as part of the 2024.6.0 update.

**Comment by MarcinZiabek** [comment]
This bug should be fixed in the 2024.6.0 release. Could you please give it a try and confirm? 😄

**Comment by rverberne-carapax** [comment]
The bug is fixed using the reproduction code. I dont have time to verify with the full code. I consider this fixed. Thx!

**Comment by rverberne-carapax** [comment]
Another thing: I noticed a change in behavior when using 2024.6.0. A table with only a head (no rows) does not get displayed anymore. In version 2024.3.10 the table head does get displayed. I dont know if this is a related, just saying.

**Comment by MarcinZiabek** [comment]
Indeed! Thank you for noticing this issue! 😄 

It was a consequence of another decision related to enhanced layout measurement logic. However, I agree that this behavior does not apply to the `Table` component. It should be fixed in the `2024.6.1` release.
